common/rpctypes/utils.go

```diff
@@ -1,70 +1 @@
 package rpctypes
-
-import (
-	"bytes"
-	"context"
-	"encoding/json"
-	"fmt"
-	"io"
-	"net"
-	"net/http"
-	"time"
-)
-
-var (
-	contentTypeJSON   = "application/json"
-	dialTimeout       = 60 * time.Second
-	httpClientTimeout = 30 * time.Minute
-	callTimeout       = 30 * time.Minute
-
-	transport = &http.Transport{
-		Dial: (&net.Dialer{
-			Timeout: dialTimeout,
-		}).Dial,
-	}
-	httpClient = &http.Client{
-		Transport: transport,
-		Timeout:   httpClientTimeout,
-	}
-)
-
-// PostRPC posts a JSON-RPC call to the given endpoint.
-func PostRPC(endpoint, method, params string) (*Response, error) {
-	data := []byte(`{"jsonrpc":"2.0","method":"` + method + `","params":` + params + `,"id":0}`)
-	buf := &bytes.Buffer{}
-	_, err := buf.Write(data)
-	if err != nil {
-		return nil, err
-	}
-
-	r, err := http.NewRequest("POST", endpoint, buf)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create HTTP request: %w", err)
-	}
-	r.Header.Set("Content-Type", contentTypeJSON)
-
-	ctx, cancel := context.WithTimeout(context.Background(), callTimeout)
```

Comment:
The old version had a base context that wasn't using the existing context of the client object that was invoking it.